Added saving admin even after room restarts

Anooxy17 opened the pull request:
It's tested. Maybe needs to optimize, you can check, greetings.

Anooxy17 committed:
Added saving admin even after room restarts

Anooxy17 committed:
Forgotten db file

jakjus commented:
Thanks for your contribution. Code looks good to me. I just wonder, what if I just want to give someone admin to take care of players for 5 minutes, but that person exits room? If he comes back after few days, he would still have the admin (and possibly I'm not in the room to revoke it).

How can a room admin revoke his admin privileges then? He would have to open .sqlite manually, which most of the people cannot do. Also, it cannot be done on database thats currently running transaction, so it would require turning off the room to revoke the admin.

Also, it could be counter-intuitive. People expect that restarting a room process will reset the admins so that only person who has process control can reassign admins. 

Think of a league room (there are leagues that use this script already). League admin gives temporary admin to the team captains and turns off the room after game. During next league match, when previous game's captains want to spectate, they would reappear with admin.

Anooxy17 commented:
You're right, I will try to add today removing player admin from database by auth or something.

If someone wants temporary (for 5 minutes), just give him admin by single 
click haxball admin.

I was hosting a lot of league rooms, admins are for captains, if you want to give all of them, the function about giving admins is completely wrong, if its still for "league admins", it should stay at it is, can add option into settings saveAdmins or anything, I was mostly thinking about pub room.

jakjus commented:
Ah, my mistake, this PR is only to reapply admins to people who got admin through !admin command. I overlooked that.

Sounds good.

Anooxy17 committed:
Updated command.ts file with new !deladmin <auth> and !ladmin commands

I've added 2 new commands due to our previous chat
!deladmin <auth> that can remove player admin by auth
!ladmin with admin list, syntax name - [auth]

Yes, I know it's gonna be problematic for now until there's no master rank or something for "owners".

Anooxy17 committed:
Update db.ts

Updated db file (support for deladmin and ladmin commands)

Anooxy17 commented:
I've updated code, will prepare master rank soon aswell, so there wont be problem with random players removing, but still its open-source, so people can choose how they want to remove admins, etc. Or maybe someone has any other idea ;)

jakjus commented on src/command.ts:
```diff
@@ -42,12 +43,43 @@ const adminLogin = (p: PlayerAugmented, args: string[]) => {
+    );
+    return; 
+  }
+  await getAdminsList(p);
```
Output will be too long if it lists all admin's auths. It should be truncated into multiple messages.

jakjus reviewed (commented):
I'm still not sure about this... I'm not sure if it's a good solution for majority of people. Just a bit counterintuitive, also that the main host can get admin deleted (the control will be overtaken) after he shares admin password. And to retake it he would have to grant himself admin again and be FASTER than the other admin that could revoke it from him immediately. Especially that main admin possible would have the same auth, so he could spam `!deladmin auth` even after room restart. I need to think a bit longer about this one. 

While it works well on your pub I don't know if it's a good idea to merge it.

Anooxy17 commented:
You're right, we trust our people at server, so even with 15 admins there, still none is abusing it. But you can keep it in mind to add stable admins etc, command is still not the best option, same as my PR here. One leak and room can be gone 🫣
